zerver/lib/actions.py

```diff
@@ -2161,7 +2165,7 @@ def do_add_submessage(
     ums = UserMessage.objects.filter(message_id=message_id)
     target_user_ids = [um.user_profile_id for um in ums]
 
-    send_event(realm, event, target_user_ids)
+    transaction.on_commit(lambda: send_event(realm, event, target_user_ids))
```

Comment:
I've sorta forgotten the background here, but is the failure mode obvious if we fail to move the `send_event` call into `transaction.on_commit()`?  (E.g. tests fail).  If not, we may want to think about how to ensure that's the case.

Reply:
No, if someone were to move `send_event` outside `on_commit` later on, none of the test would fail.
I'm not sure how we can test this though 🤔 Maybe mocking `send_event` to throw an error?

Reply:
Update: Mocking `send_event` is actually useful (https://github.com/zulip/zulip/pull/18559#issuecomment-854667223)